client/src/pages/Signup/index.js

```diff
@@ -1,5 +1,234 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import { useHistory, Link } from 'react-router-dom';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Avatar from '@material-ui/core/Avatar';
+import axios from 'axios';
+import { Alert } from '@material-ui/lab';
+import { SIGNIN_PAGE } from '../../utils/router.constant';
+import { InputField, Loading } from '../../component';
+import useStyles from './style';
+import {
+  updateAndValidateInput,
+  firstNameSchema,
+  lastNameSchema,
+  emailSchema,
+  passwordSchema,
+  confirmPasswordSchema,
+} from '../../utils';
 
-const Signup = () => <div>Hi from Signup</div>;
+const Signup = () => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [lastNameError, setLastNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(false);
+  const [passwordsMismatchError, setPasswordsMismatchError] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+  const [Error, setError] = useState('');
+
+  const clear = () => {
+    setFirstName('');
+    setLastName('');
+    setEmail('');
+    setPassword('');
+    setConfirmPassword('');
+    setError(null);
```

Comment:
suggestion :
I think default value is better to be the same type. 
```suggestion
    setError('');
```